core/arch/aarch64/proc.c

```diff
@@ -217,6 +226,73 @@ proc_init_arch(void)
 #define GET_FEAT_VAL(FEATURE) (((ushort)FEATURE) & 0x000F)
 #define GET_FEAT_NSFLAG(FEATURE) ((((ushort)FEATURE) & 0x8000) >> 15)
 
+#if defined(BUILD_TESTS) || defined(STANDALONE_DECODER)
+void
+proc_set_feature(feature_bit_t f, bool enable)
+{
+    uint64 *freg_val = 0;
+    ushort feat_nibble = GET_FEAT_NIBPOS(f);
+    uint64 feat_nsflag = GET_FEAT_NSFLAG(f);
+    uint64 feat_val = GET_FEAT_VAL(f);
+
+    feature_reg_idx_t feat_reg = GET_FEAT_REG(f);
+    switch (feat_reg) {
+    case AA64ISAR0: {
+        freg_val = &cpu_info.features.flags_aa64isar0;
+        break;
+    }
+    case AA64ISAR1: {
+        freg_val = &cpu_info.features.flags_aa64isar1;
+        break;
+    }
+    case AA64PFR0: {
+        freg_val = &cpu_info.features.flags_aa64pfr0;
+        break;
+    }
+    case AA64MMFR1: {
+        freg_val = &cpu_info.features.flags_aa64mmfr1;
+        break;
+    }
+    case AA64DFR0: {
+        freg_val = &cpu_info.features.flags_aa64dfr0;
+        break;
+    }
+    case AA64ZFR0: {
+        freg_val = &cpu_info.features.flags_aa64zfr0;
+        break;
+    }
+    default: CLIENT_ASSERT(false, "proc_has_feature: invalid feature register");
+    }
+
+    /* Clear the current feature state */
```

Comment:
nit: Prefer end punctuation for new comments
More below

Reply:
Done.